qiskit/ignis/mitigation/expval/complete_mitigator.py

```diff
@@ -0,0 +1,179 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Full-matrix measurement error mitigation generator.
+"""
+from typing import Optional, List, Dict, Tuple
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from .utils import (counts_probability_vector, _expval_with_stddev)
+from .base_meas_mitigator import BaseExpvalMeasMitigator
+
+
+class CompleteExpvalMeasMitigator(BaseExpvalMeasMitigator):
+    """N-qubit measurement error mitigator.
+
+    This class can be used with the
+    :func:`qiskit.ignis.mitigation.expectation_value` function to apply
+    measurement error mitigation of general N-qubit measurement errors
+    when calculating expectation values from counts. Expectation values
+    can also be computed directly using the :meth:`expectation_value` method.
+
+    For measurement mitigation to be applied the mitigator should be
+    calibrated using the
+    :func:`qiskit.ignis.mitigation.expval_meas_mitigator_circuits` function
+    and :class:`qiskit.ignis.mitigation.ExpvalMeasMitigatorFitter` class with
+    the ``'complete'`` mitigation method.
+    """
+
+    def __init__(self, amat: np.ndarray):
+        """Initialize a TensorMeasurementMitigator
+
+        Args:
+            amat (np.array): readout error assignment matrix.
+        """
+        self._num_qubits = int(np.log2(amat.shape[0]))
+        self._assignment_mat = amat
+        self._mitigation_mats = {}
+
+    def expectation_value(self,
+                          counts: Dict,
+                          diagonal: Optional[np.ndarray] = None,
+                          qubits: Optional[List[int]] = None,
+                          clbits: Optional[List[int]] = None,
+                          ) -> Tuple[float, float]:
+        r"""Compute the mitigated expectation value of a diagonal observable.
+
+        This computes the mitigated estimator of
+        :math:`\langle O \rangle = \mbox{Tr}[\rho. O]` of a diagonal observable
+        :math:`O = \sum_{x\in\{0, 1\}^n} O(x)|x\rangle\!\langle x|`.
+
+        Args:
+            counts: counts object
+            diagonal: Optional, the vector of diagonal values for suming the
+                      expectation value. If ``None`` the the default value is
+                      :math:`[1, -1]^\otimes n`.
+            qubits: Optional, the measured physical qubits the count
+                    bitstrings correspond to.
+            clbits: Optional, marginalize counts to just these bits.
+
+        Returns:
+            (float, float): the expectation value and standard deviation.
+
+        Raises:
+            QiskitError: if input arguments are invalid.
+
+        Additional Information:
```

Comment:
Don't you want to write all this in only one place?

Reply:
I want it to show up in the API docs for each class, which means I have to copy paste it (the base class isn't actually listed in the API docs since a user never uses it directly).